src/etl/ReportingETL.cpp

```diff
@@ -826,7 +826,7 @@ ReportingETL::monitor()
                  << "Starting monitor loop. sequence = " << nextSequence;
     while (true)
     {
-        if (auto rng = backend_->hardFetchLedgerRangeNoThrow();
+        if (auto rg = backend_->hardFetchLedgerRangeNoThrow();
```

Comment:
define rg , but not use it?

Reply:
if we had uses of `rng` after this line (in the scope of `if`) then it should now be `rg` everywhere in there. 
You seem to have made a bug. You are getting the new `rg` but checking the old one. I fear that there is going to be many of those in other fixes you did - please go check all your changes super thoroughly, otherwise we might end up with multiple bugs added to the codebase.

```
if (auto rg = backend_->hardFetchLedgerRangeNoThrow(); rg && rg->maxSequence >= nextSequence) {
    publishLedger(nextSequence, {});
    ++nextSequence;
}
```

Reply:
Functionally it makes no difference as to whether rng (by removing auto to reference the variable that already exists within method scope) or rg (shadowed param solution) is used because rng in the original code is not referenced further in the while loop aside the if condition

The functional difference of the while loop is to monitor the ledger sequence. The difference is that with rg, you create a new copy for the sequence data whereas with rng you're overwriting the old variable used to store it, which wasn't utilized in there anyway